src/solver/td3.ml

```diff
@@ -368,22 +409,24 @@ module Base =
       and simple_solve l x y =
         if tracing then trace "sol2" "simple_solve %a (rhs: %b)" S.Var.pretty_trace y (Hooks.system y <> None);
         if Hooks.system y = None then (init y; HM.replace stable y (); HM.find rho y) else
-        if not space || HM.mem wpoint y then (solve y Widen; HM.find rho y) else
+          (* TODO: should td_space store information for widening points with remaining gas? *)
+        if not space || HM.mem wpoint_gas y then (solve y Widen; HM.find rho y) else
         if HM.mem called y then (init y; HM.remove l y; HM.find rho y) else (* TODO: [HM.mem called y] is not in the TD3 paper, what is it for? optimization? *)
-        (* if HM.mem called y then (init y; let y' = HM.find_default l y (S.Dom.bot ()) in HM.replace rho y y'; HM.remove l y; y') else *)
+          (* if HM.mem called y then (init y; let y' = HM.find_default l y (S.Dom.bot ()) in HM.replace rho y y'; HM.remove l y; y') else *)
         if cache && HM.mem l y then HM.find l y
         else (
           HM.replace called y ();
           let eqd = eq y (eval l x) (side ~x) in
           HM.remove called y;
-          if HM.mem wpoint y then (HM.remove l y; solve y Widen; HM.find rho y)
+          (* TODO: is should_widen correct here? *)
+          if should_widen y then (HM.remove l y; solve y Widen; HM.find rho y)
```

Comment:
Same `HM.mem wpoint` thing here. This seems to be just starting a new widening phase.